CMakeLists.txt

```diff
@@ -60,12 +64,23 @@ include_directories(osal)
 include_directories(osal/${OS})
 include_directories(oshw/${OS})
 
-add_library(soem STATIC
-  ${SOEM_SOURCES}
-  ${OSAL_SOURCES}
-  ${OSHW_SOURCES}
-  ${OSHW_EXTRA_SOURCES})
-target_link_libraries(soem ${OS_LIBS})
+
+
+if(BUILD_SHARED_LIBS)
+  add_library(soem SHARED
+    ${SOEM_SOURCES}
+    ${OSAL_SOURCES}
+    ${OSHW_SOURCES}
+    ${OSHW_EXTRA_SOURCES})
+  target_link_libraries(soem ${OS_LIBS})
```

Comment:
set_property(TARGET soem PROPERTY POSITION_INDEPENDENT_CODE ON)?

Reply:
Ignore that, seems -fPIC is set automatically when building SHARED so this is not needed.

Reply:
Copied